Marlin/src/feature/touch/xpt2046.cpp

```diff
@@ -77,14 +81,24 @@ uint8_t XPT2046::read_buttons() {
   const uint16_t x = uint16_t(((uint32_t(getInTouch(XPT2046_X))) * tsoffsets[0]) >> 16) + tsoffsets[1],
                  y = uint16_t(((uint32_t(getInTouch(XPT2046_Y))) * tsoffsets[2]) >> 16) + tsoffsets[3];
   if (!isTouched()) return 0; // Fingers must still be on the TS for a valid read.
-
-  if (y < 175 || y > 234) return 0;
-
-  return WITHIN(x,  14,  77) ? EN_D
-       : WITHIN(x,  90, 153) ? EN_A
-       : WITHIN(x, 166, 229) ? EN_B
-       : WITHIN(x, 242, 305) ? EN_C
-       : 0;
+  
+  // Touch within the button area simulates an encoder button
+  if (y > BUTTON_AREA_TOP && y < BUTTON_AREA_BOT)
+    return WITHIN(x,  14,  77) ? EN_D
+         : WITHIN(x,  90, 153) ? EN_A
+         : WITHIN(x, 166, 229) ? EN_B
+         : WITHIN(x, 242, 305) ? EN_C
+         : 0;
+
+  // Column and row above BUTTON_AREA_TOP
+  int8_t col = (x % (LCD_FULL_PIXEL_WIDTH) ) * (LCD_WIDTH ) / (LCD_FULL_PIXEL_WIDTH),
+         row = (y % (LCD_FULL_PIXEL_HEIGHT)) * (LCD_HEIGHT) / (BUTTON_AREA_TOP);
+
+  row = (LCD_HEIGHT) - row - 1; // TODO: Can LCD or sensor be inverted?
+
+  // Send the touch to the UI (which will simulate the encoder wheel)
+  MarlinUI::screen_click(row, col, x, y);
+  return 0;
```

Comment:
What happens if you return `EN_C` here instead of 0? Does the correct menu item get activated? If not, we can fix that!

Reply:
Do you want simulate the click? I didnt try, but seems will work. But I liked more the touch acting like selection.

Reply:
But if the touch is on the already-selected item…

Reply:
It will be nice. But, in this case, who needs to decide if clicks or not is the UI part, right? Right there we can check if the current line is the same one touched and send a click.

I will work on that in a minute.

Reply:
the code already do multiple "clicks" if you move the finger and stay pressed...